service/history/execution/context.go

```diff
@@ -512,8 +518,10 @@ func (c *contextImpl) ConflictResolveWorkflowExecution(
 			return err
 		}
 		if len(resetWorkflow.WorkflowRequests) != 0 && len(newWorkflow.WorkflowRequests) != 0 {
-			// TODO(CDNC-8519): convert it to an error after verification in production
-			c.logger.Error("Workflow reqeusts are only expected to be generated from one workflow for ConflictResolveWorkflowExecution")
+			if c.shard.GetConfig().EnableStrongIdempotencySanityCheck(domain) {
+				return &types.InternalServiceError{Message: "workflow requests are only expected to be generated from either reset workflow or continue-as-new workflow for ConflictResolveWorkflowExecution"}
+			}
+			c.logger.Error("workflow requests are only expected to be generated from either reset workflow or continue-as-new workflow for ConflictResolveWorkflowExecution")
```

Comment:
let's include workflow request counts in the error message as well as maybe request ids etc. to help with troubleshooting of this case